### ISSUE TITLE
unknown format not supported for properties

### ISSUE THREAD

**Issue by yzatslinear** [RobWin/swagger2markup#13]
My definitions contains string properties of "user-defined" formats, e.g. "email" below. According to 2.0 spec, this is valid. 

```
"Order":{
 "properties": {
                "emailAddr": {
                    "type": "string",
                    "format": "email"
                }
            }
}
```

When this is encountered, I receive an exception:

```
java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:53)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NullPointerException
	at io.github.robwin.swagger2markup.builder.document.DefinitionsDocument.definition(DefinitionsDocument.java:110)
	at io.github.robwin.swagger2markup.builder.document.DefinitionsDocument.definitions(DefinitionsDocument.java:73)
	at io.github.robwin.swagger2markup.builder.document.DefinitionsDocument.build(DefinitionsDocument.java:57)
	at io.github.robwin.swagger2markup.Swagger2MarkupConverter.buildDocuments(Swagger2MarkupConverter.java:92)
	at io.github.robwin.swagger2markup.Swagger2MarkupConverter.intoFolder(Swagger2MarkupConverter.java:72)
	at com.mycompany.app.App.main(App.java:38)
	... 6 more

```

**Comment by RobWin** [comment]
Thx. Yes the Converter is not Swagger Spec complete.
I will have a look at it.

**Comment by yzatslinear** [comment]
I used 0.2.3

**Comment by RobWin** [comment]
I have added your snippet to:

```
"Order": {
            "properties": {
                "id": {
                    "type": "integer",
                    "format": "int64"
                },
                "petId": {
                    "type": "integer",
                    "format": "int64"
                },
                "quantity": {
                    "type": "integer",
                    "format": "int32"
                },
                "shipDate": {
                    "type": "string",
                    "format": "date-time"
                },
                "status": {
                    "type": "string",
                    "description": "Order Status"
                },
                "complete": {
                    "type": "boolean"
                },
                "emailAddr": {
                    "type": "string",
                    "format": "email"
                }
            }
```

And it seems that the official Swagger parser does not add the emailAddr to the Swagger model:

![image](https://cloud.githubusercontent.com/assets/5815046/7175767/81eefd7c-e418-11e4-936b-31405225e30a.png)

I've fixed the NullPointerException issue, but if you have a Model like:
```
"Order":{
 "properties": {
                "emailAddr": {
                    "type": "string",
                    "format": "email"
                }
            }
}
```

The properties Map is null:

![image](https://cloud.githubusercontent.com/assets/5815046/7175794/cf5db68e-e418-11e4-9c1c-315040ac8a35.png)

Could you please open an issue against the Swagger parser: https://github.com/swagger-api/swagger-parser

**Comment by RobWin** [comment]
@yzatslinear Yes it is definitely a Swagger parser issue.
See swagger-api/swagger-parser#37

**Comment by RobWin** [comment]
NullPointerException issue fixed in v0.2.4
